cms/djangoapps/contentstore/models.py

```diff
@@ -153,16 +157,25 @@ def upstream_context_title(self) -> str:
     def filter_links(
         cls,
         **link_filter,
-    ) -> QuerySet["EntityLinkBase"]:
+    ) -> QuerySet["EntityLinkBase"] | list["EntityLinkBase"]:
         """
         Get all links along with sync flag, upstream context title and version, with optional filtering.
+
+        `use_top_level_parents` is an special filter, replace any result with the top-level parent if exists.
+        Example: We have linkA and linkB with top-level parent as linkC, and linkD without top-level parent.
+        After all other filters:
+        Case 1: `use_top_level_parents` is False, the result is [linkA, linkB, linkC, linkD]
+        Case 2: `use_top_level_parents` is True, the result is [linkC, linkD]
```

Comment:
I'm not a big fan of hiding this "special filter" in the `link_filter` kwargs  (because it's really a boolean parameter for the function - it does more than just remove some results; it replaces them). I think it will be cleaner if you just make it an explicit parameter like this:

```python
def filter_links(cls, *, use_top_level_parents=False, **link_filter):
```
That should also be fully backwards-compatible.

Reply:
Updated https://github.com/openedx/edx-platform/pull/37076/commits/e77a67f214681adbbb09a5a58e07b8a1c2800e57